fili-core/src/main/java/com/yahoo/bard/webservice/web/DataApiRequest.java

```diff
@@ -296,6 +309,87 @@ public DataApiRequest(
     }
 
     /**
+     * To check whether dateTime column request is first one in the sort list or not.
+     *
+     * @param sortColumns  map of columns and its direction
+     *
+     * @return True if dateTime column is first one in the sort list. False otherwise
+     */
+    protected Boolean isDateTimeFirstSortField(Map<String, SortDirection> sortColumns) {
```

Comment:
It's strange that we're talking about this method doing things in a list-y way, yet it takes a map. We should probably make this method take a list, since it cares about position.

At the very least, if order matters (since we convert the map to a list), we _definitely_ need to make the Map ordered in the method signature, rather than just taking a Map. (eg. HashMap will _not_ preserve order here and will not work as expected, causing irritating bugs)